### PR TITLE
Yann/test/rfid/deep sleep rfidkit functional tests

### PR DESCRIPTION
**Pull request by YannLocatelli**
# Requirements

- [x] #1366
- [ ] #1391
- [x] #1386

### CODE REVIEW

**Comment by YannLocatelli** [comment]
Missing card on robot of functional test

File: tests/functional/tests/deep_sleep_rfid_kit/suite_rfid_kit.cpp [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+	scenario("enabled/disable deepsleep then magic card detected") = [] {
+		given("rfid is in default configuration") = [] {
+			auto cardDetected = false;
+			rfidkit.onTagActivated([&cardDetected](const MagicCard &card) { cardDetected = true; });
+
+			when("I wait") = [&] {
+				cardDetected = false;
+				rtos::ThisThread::sleep_for(1s);
+
+				then("I expect TO detect card") = [&cardDetected] { expect(cardDetected); };
```
Comment (ladislas) [inline tests/functional/tests/deep_sleep_rfid_kit/suite_rfid_kit.cpp:100]:
```suggestion
				then("I expect card TO BE detected") = [&cardDetected] { expect(cardDetected); };
```

File: tests/functional/tests/deep_sleep_rfid_kit/suite_rfid_kit.cpp [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+
+					expect(status.test_check_ok);
+				};
+			};
+
+			when("I wait") = [&] {
+				cardDetected = false;
+				rtos::ThisThread::sleep_for(1s);
+
+				then("I expect TO NOT detect card") = [&cardDetected] { expect(not cardDetected); };
```
Comment (ladislas) [inline tests/functional/tests/deep_sleep_rfid_kit/suite_rfid_kit.cpp:119]:
```suggestion
				then("I expect card TO NOT BE detected") = [&cardDetected] { expect(not cardDetected); };
```

File: tests/functional/tests/deep_sleep_rfid_kit/suite_rfid_kit.cpp [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+
+					expect(not status.test_check_ok);
+				};
+			};
+
+			when("I wait") = [&] {
+				cardDetected = false;
+				rtos::ThisThread::sleep_for(1s);
+
+				then("I expect TO detect card") = [&cardDetected] { expect(cardDetected); };
```
Comment (ladislas) [inline tests/functional/tests/deep_sleep_rfid_kit/suite_rfid_kit.cpp:138]: same as above

**Review by ladislas — CHANGES_REQUESTED** [review]
lGTM modulo le wording pour les tests rfid

**Pull request merged**